packages/server/modules/core/utils/dbNotificationListener.ts

```diff
@@ -1,33 +1,78 @@
-import { MaybeAsync, Optional } from '@speckle/shared'
-import { dbNotificationLogger, moduleLogger } from '@/logging/logging'
+import { MaybeAsync, Optional, md5 } from '@speckle/shared'
+import { dbNotificationLogger } from '@/logging/logging'
 import { knex } from '@/modules/core/dbSchema'
 import * as Knex from 'knex'
 import * as pg from 'pg'
-
-/**
- * TODO: This currently will emit duplicate events when there are multiple server instances running. Not a big deal currently while there aren't that many events,
- * but we need to figure this out
- */
+import { createRedisClient } from '@/modules/shared/redis/redis'
+import { getRedisUrl } from '@/modules/shared/helpers/envHelper'
+import Redis from 'ioredis'
+import { LogicError } from '@/modules/shared/errors'
 
 export type MessageType = { channel: string; payload: string }
 export type ListenerType = (msg: MessageType) => MaybeAsync<void>
 
 let shuttingDown = false
 let connection: Optional<pg.Connection> = undefined
+let redisClient: Optional<Redis> = undefined
+
 const listeners: Record<string, { setup: boolean; listener: ListenerType }> = {}
+const lockName = 'server_postgres_listener_lock'
+
+function getMessageId(msg: MessageType) {
+  const str = JSON.stringify(msg)
+  return md5(str)
+}
+
+async function getTaskLock(taskId: string) {
+  if (!redisClient) {
+    throw new LogicError(
+      'Unexpected failure! Attempting to get task lock before redis client is initialized'
+    )
+  }
+
+  const lockKey = `${lockName}:${taskId}`
+  const lock = await redisClient.set(lockKey, '1', 'EX', 60 * 60, 'NX')
+  const releaseLock = async () => {
+    if (!redisClient) {
+      throw new LogicError(
+        'Unexpected failure! Attempting to release task lock before redis client is initialized'
+      )
+    }
+    await redisClient.del(lockKey)
+  }
+  return lock ? releaseLock : null
+}
 
-function messageProcessor(msg: MessageType) {
+async function messageProcessor(msg: MessageType) {
   const listener = listeners[msg.channel]
-  dbNotificationLogger.info(
-    {
-      ...msg,
-      listenerRegistered: !!listener
-    },
-    'Message received'
+  const messageId = getMessageId(msg)
+
+  const logPayload = {
+    ...msg,
+    listenerRegistered: !!listener,
+    messageId
+  }
+  dbNotificationLogger.debug(
+    logPayload,
+    'Message #{messageId} received in channel {channel}'
   )
   if (!listener) return
 
-  return listener.listener(msg)
+  // Only process if lock acquired
+  const unlock = await getTaskLock(messageId)
+  if (unlock) {
+    dbNotificationLogger.info(
```

Comment:
Lets lower this to debug.

Reply:
do we really want all of those to be debug level? if we're capturing requests w/ INFO, I would think that we would want the same for incoming DB events, at least the ones that are actually processed, not skipped